cli/azd/internal/appdetect/java_test.go

```diff
@@ -0,0 +1,154 @@
+package appdetect
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestJavaDetector_Language(t *testing.T) {
+	jd := &javaDetector{}
+	if jd.Language() != Java {
+		t.Errorf("expected language to be Java, got %v", jd.Language())
+	}
+}
+
+func TestJavaDetector_DetectProject_WithPomXml(t *testing.T) {
```

Comment:
If it make sense to you: this is covered by the acceptance tests in `appdetect_test.go`, so we could consider removing it.